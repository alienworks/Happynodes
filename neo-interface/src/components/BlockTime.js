
import React, { Component } from 'react'
import { connect } from 'react-refetch'
import config from './config'

class BlockTime extends Component {
    render() {
        const {blocktime} = this.props

        if (blocktime.pending) {
            return (
               <p></p>
            );
        }
        
        if (blocktime.value === null || blocktime.value === 'undefined'){
            return (
                <div className="top-block yellow">
                <h2>0s</h2>
                <p>AVG BLOCK TIME</p></div>
            );
        }
        return (
            <div className="top-block yellow">
            <h2>{blocktime.value.blocktime}s</h2>
            <p>AVG BLOCK TIME</p></div>
        );
    }
}
  

export default connect( (props)=> ({
    blocktime: {
        url: config.api_url.concat(`/blocktime`),
        refreshInterval: 3000
    }
}))(BlockTime)