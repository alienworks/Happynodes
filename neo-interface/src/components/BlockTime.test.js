import React from 'react';
import ReactDOM from 'react-dom';
import BlockTime from './BlockTime';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BlockTime />, div);
    ReactDOM.unmountComponentAtNode(div);
});


describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<BlockTime blocktime={{value:{blocktime:1234}}} />).toJSON();

        expect(tree).toMatchSnapshot();

    });
});
  