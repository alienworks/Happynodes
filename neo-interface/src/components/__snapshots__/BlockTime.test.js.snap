// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  className="top-block yellow"
>
  <h2>
    1234
    s
  </h2>
  <p>
    AVG BLOCK TIME
  </p>
</div>
`;
